src/mlpack/methods/edge_boxes/feature_extraction.hpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file feature_extraction.hpp
+ * @author Nilay Jain 
+ *
+ * Feature Extraction for the edge_boxes algorithm.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+//#define INF 999999.9999
+//#define EPS 1E-20
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace structured_tree {
+
+template <typename MatType = arma::mat, typename CubeType = arma::cube>
+class StructuredForests
+{
+
+ public:
+
+  static constexpr double eps = 1e-20;
+
+  std::map<std::string, size_t> options;
+  
+  StructuredForests(const std::map<std::string, size_t> inMap);
+  
+  MatType LoadData(MatType const &images, MatType const &boundaries,\
+     MatType const &segmentations);
+
+  void PrepareData(MatType const &InputData);
+
+  arma::vec GetFeatureDimension();
+  
+  arma::vec DistanceTransform1D(arma::vec const &f, const size_t n,\
+                                 const double inf);
+  
```

Comment:
Another picky comment, please use prefix const, also I know it's a matter of style, but I prefer to use ``type&parameter`` instead of ``type&parameter``. For more details: https://github.com/mlpack/mlpack/pull/523/files/c92dd2fc6561b1cf40efb744dcc9233bc3675c91#r60094845